### PR TITLE
Simplify RapidCheck configure

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

No more `RAPIDCHECK_HEADERS`!

# Context

Flake lock file updates:

• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/86501af7f1d51915e6c335f90f2cab73d7704ef3' (2024-01-11)
  → 'github:NixOS/nixpkgs/a1982c92d8980a0114372973cbdfe0a307f1bdea' (2024-01-12)

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: flake.lock [hunk 1]
```diff
@@ -34,11 +34,11 @@
...
     "nixpkgs": {
       "locked": {
-        "lastModified": 1704018918,
-        "narHash": "sha256-erjg/HrpC9liEfm7oLqb8GXCqsxaFwIIPqCsknW5aFY=",
+        "lastModified": 1704982786,
+        "narHash": "sha256-K5eJHmL1/kev6WuqyqqbS1cdNnSidIZ3jeqJ7GbrYnQ=",
         "owner": "NixOS",
         "repo": "nixpkgs",
-        "rev": "2c9c58e98243930f8cb70387934daa4bc8b00373",
+        "rev": "a1982c92d8980a0114372973cbdfe0a307f1bdea",
```
Comment (Ericson2314) [inline flake.lock:41]: The channel didn't actually bump yet, but I strongly expect https://hydra.nixos.org/eval/1803537#tabs-inputs to pass soon.

**Comment by edolstra** [comment]
Looks like the lock file update is incorrect.

We should wait until the channel bumps, otherwise users who track the Nix flake and the Nixpkgs 23.05 channel won't be able to upgrade their systems.

File: flake.lock (hunk 1 above)
Comment (Ericson2314) [reply flake.lock:41]: Now it did

**Review by roberth — APPROVED** [review]
Nice

**Pull request merged**